archibald/cogs/alias.py

```diff
@@ -1,57 +1,51 @@
-import discord
 import json
-from discord import option
 from discord.ext import commands
+from discord.app_commands import describe
+from discord.ext.commands.context import Context
 
-class link_alias(commands.Cog):
+class LinkAlias(commands.Cog):
     def __init__(self, bot):
         self.bot = bot
-        self.link_map = json.loads(open("archibald/cogs/links.json").read())['links'];
-        self.add_alias_whitelist = [
-            332935845004705793, 
-            655144378062471173, 
-            845784118536175676, 
-            464247726381727744, 
-            192732093019717632  
-        ]
+        self.link_map: dict[str, str] = json.loads(open("src/cogs/links.json").read())['links'];
+        self.admin = 872083306668261437
+        self.mod = 1000862083639943228
 
-    @commands.Cog.listener()
-    async def on_message(self, message: discord.Message):
-        alias = message.content.split(" ")[0]
-        if f"{alias}" in self.link_map.keys():
-            await message.channel.send(self.link_map[alias])
+    async def cog_check(self, ctx: Context):
+        if (ctx.command.name == "addalias"):
+            role = ctx.author.get_role(self.admin) or ctx.author.get_role(self.mod)
+            return (role != None)
+        return True
 
```

Comment:
oh hm this check should happen in the command itself instead of the whole cog, didn't realize this before, sorry :D